### ISSUE TITLE
How to understand the DeepFluoro  isocenter? 

### ISSUE THREAD

**Issue by BigPandaCPU** [eigenvivek/DiffPose#38]
Hi Vivek,

Thanks for sharing the code for your awesome work ! I'm confused by the caclutation of DeepFluoro isocenter pose.
The source cobe below:
   deepfluoro.py line48
    # Get the isocenter pose (AP viewing angle at volume isocenter)
    isocenter_rot = torch.tensor([[torch.pi / 2, 0.0, -torch.pi / 2]])
    isocenter_xyz = torch.tensor(self.volume.shape) * self.spacing / 2
    isocenter_xyz = isocenter_xyz.unsqueeze(0)
    self.isocenter_pose = RigidTransform(
        isocenter_rot, isocenter_xyz, "euler_angles", "ZYX"
    )

How to understand the isocenter_rot =  torch.tensor([[torch.pi / 2, 0.0, -torch.pi / 2]])???

From your code,the initial DRR's source is (sdd/2, 0, 0), target is (-sdd/2, 0, 0), to the AP view, we only need to rotate around the Z  axis, why there rotate around Z and X?

**Comment by BigPandaCPU** [comment]
> Hi Vivek,
> 
> Thanks for sharing the code for your awesome work ! I'm confused by the caclutation of DeepFluoro isocenter pose. The source cobe below: deepfluoro.py line48 # Get the isocenter pose (AP viewing angle at volume isocenter) isocenter_rot = torch.tensor([[torch.pi / 2, 0.0, -torch.pi / 2]]) isocenter_xyz = torch.tensor(self.volume.shape) * self.spacing / 2 isocenter_xyz = isocenter_xyz.unsqueeze(0) self.isocenter_pose = RigidTransform( isocenter_rot, isocenter_xyz, "euler_angles", "ZYX" )
> 
> How to understand the isocenter_rot = torch.tensor([[torch.pi / 2, 0.0, -torch.pi / 2]])???
> 
> From your code,the initial DRR's source is (sdd/2, 0, 0), target is (-sdd/2, 0, 0), to the AP view, we only need to rotate around the Z axis, why there rotate around Z and X?


I got the answer, because you have switched the X and Y axes. I use ITK-snap to read the data you provided data/crx, the figure shows below:
![01](https://github.com/user-attachments/assets/ff2a2cab-1f57-4ee7-b720-32afd7f117fe)

while I use ITK-snap to read the data (drr.py line 111), 
![03](https://github.com/user-attachments/assets/6ed5231e-2b0e-45b5-8c8d-8993631effe2)

the figure shows below:
![02](https://github.com/user-attachments/assets/52496272-c9ed-427c-92b4-2fda9bb4f188)

I found that the data's axis of X and Y has switched!

**Comment by BigPandaCPU** [comment]
> Hi Vivek,
> 
> Thanks for sharing the code for your awesome work ! I'm confused by the caclutation of DeepFluoro isocenter pose. The source cobe below: deepfluoro.py line48 # Get the isocenter pose (AP viewing angle at volume isocenter) isocenter_rot = torch.tensor([[torch.pi / 2, 0.0, -torch.pi / 2]]) isocenter_xyz = torch.tensor(self.volume.shape) * self.spacing / 2 isocenter_xyz = isocenter_xyz.unsqueeze(0) self.isocenter_pose = RigidTransform( isocenter_rot, isocenter_xyz, "euler_angles", "ZYX" )
> 
> How to understand the isocenter_rot = torch.tensor([[torch.pi / 2, 0.0, -torch.pi / 2]])???
> 
> From your code,the initial DRR's source is (sdd/2, 0, 0), target is (-sdd/2, 0, 0), to the AP view, we only need to rotate around the Z axis, why there rotate around Z and X?

I understand it, the rotation order is first rotate around X axis, then rotate around Z axis. Rotate around X axis in order to get the DRR image in nomral order, If don't  rotate around X axis, the DRR image is like below. 
![04](https://github.com/user-attachments/assets/b432df24-52f1-4a4e-be7d-4a2b0ff58ea0)
